docs/docs/spec.md

```diff
@@ -615,18 +615,28 @@ The same-type relation is *not* defined to be reflexive or transitive.
 > For more discussion of reflexive and transitive relations, see the comments
 > under [nullness subtyping].
 
+> Compare [JLS 4.3.4]. Note that our definition of "same type" applies to all
+> kinds of augmented types, including cases like the augmented null types.
+
 ## Subtyping
 
-`A` is a subtype of `F` if both of the following conditions are met:
+`A` is a subtype of `F` if all of the following conditions are met:
 
 -   `A` is a [nullness subtype] of `F`.
 -   `A` is a subtype of `F` according to the
     [nullness-delegating subtyping rules for Java].
+-   The base type of `A` is a subtype of the base type of `F` according to
```

Comment:
I feel like this one should come first. I guess the order doesn't technically matter, but this one will usually get applied first, as you imply.

Is there any opportunity to simplify the nullness-delegating subtyping rules for Java now that it's explicit that the regular subtyping rules for Java have to apply as well?

Reply:
Sure, I've moved it first.

As for the nullness-delegating subtyping rules for Java: I think I already wrote those under the implicit assumption that we could assume that base-type subtyping held, so I don't see opportunity for simplification.

I do hope to simplify those rules someday, but it is likely to be a big project. I've filed https://github.com/jspecify/jspecify/issues/638 to at least track that beyond the "we should fix it" note in the spec.